### PR TITLE
tcp: switching to the new pool

### PR DESCRIPTION
**Pull request by alyssawilk**
Risk Level: High (new tcp pool, lots of smoke testing)
Testing: integration tests, unit tests, production validation by Pinterest
Docs Changes: n/a
Release Notes: inline

### CODE REVIEW

**Review by rgs1 — DISMISSED** [review]
We've been running this in prod for a while now (2+ weeks), looking good!

**Comment by rgs1** [comment]
> We've been running this in prod for a while now (2+ weeks), looking good!

That is, 2+ weeks since the latest fix of the last remaining known issue. Way longer than that in total, if you count cumulatively.

File: docs/root/version_history/current.rst [hunk 1]
```diff
@@ -47,6 +47,7 @@ Removed Config or Runtime
 * http: removed ``envoy.reloadable_features.overload_manager_disable_keepalive_drain_http2``; Envoy will now always send GOAWAY to HTTP2 downstreams when the :ref:`disable_keepalive <config_overload_manager_overload_actions>` overload action is active.
 * http: removed ``envoy.reloadable_features.http_match_on_all_headers`` runtime guard and legacy code paths.
 * http: removed ``envoy.reloadable_features.unify_grpc_handling`` runtime guard and legacy code paths.
+* tcp: added support for :ref:`preconnecting <v1.18.0:envoy_v3_api_msg_config.cluster.v3.Cluster.PreconnectPolicy>`. Preconnecting is off by default, but recommended for clusters serving latency-sensitive traffic.
```
Comment (ggreenway) [inline docs/root/version_history/current.rst:50]: I think this is in the wrong section. Probably belongs below in `New Features`?

**Review by ggreenway — CHANGES_REQUESTED** [review]
Exciting to see this moving forward! Minor issue with the changelog, then it's good to go.

**Review by ggreenway — APPROVED** [review]

**Pull request merged**